beacon/validator/block_builder.go

```diff
@@ -37,22 +39,24 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-// buildBlockAndSidecars builds a new beacon block.
+// BuildBlockAndSidecars builds a new beacon block.
 func (s *Service[
-	_, BeaconBlockT, _, _, BlobSidecarsT, _, _, _, _, _, _, _, SlotDataT,
-]) buildBlockAndSidecars(
+	AttestationDataT, BeaconBlockT, _, _, _, BlobSidecarsT,
+	_, _, _, _, _, _, SlashingInfoT, SlotDataT,
+]) BuildBlockAndSidecars(
 	ctx context.Context,
-	slotData SlotDataT,
-) (BeaconBlockT, BlobSidecarsT, error) {
-	var (
-		blk       BeaconBlockT
-		sidecars  BlobSidecarsT
-		startTime = time.Now()
-		g, _      = errgroup.WithContext(ctx)
-	)
+	slotData types.SlotData[ctypes.AttestationData, ctypes.SlashingInfo],
+) ([]byte, []byte, error) {
+	// as the async approach built blocks in single thread lets keep the
+	// same approach for now
+	s.blockBuilderMu.Lock()
+	defer s.blockBuilderMu.Unlock()
```

Comment:
@fridrik01 I think we don't really need this mutex anymore, since this function is only called, directly from prepareBlock?